addons/api/addon/services/sqlite.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Service from '@ember/service';
+import { PWBHost } from 'promise-worker-bi';
+
+// A mapping of columns to their expected model attributes for supported models
+// The order matters, it _must_ be the same as the order of the columns in the database.
+// The JSON data is assumed to always be last.
+export const modelMapping = {
+  target: {
+    id: 'id',
+    type: 'type',
+    name: 'name',
+    description: 'description',
+    address: 'address',
+    scope_id: 'scope.scope_id',
+    created_time: 'created_time',
+  },
+};
+
+// A list of tables that we support searching using FTS5 in SQLite.
+export const searchTables = new Set(['target']);
```

Comment:
If we end up always using or not using FTS5 at all for searching, we can remove this